documentation/src/main/asciidoc/ch07.asciidoc

```diff
@@ -1,3 +1,324 @@
 [[chapter-valueextraction]]
 == Value extraction
 
+Value extraction is the process extracting values from a container so that
+they can be validated.
+
+It is used when dealing with <<container-element-constraints,container element
+constraints>> and <<example-cascaded-validation-list,cascaded validation inside
+containers>>.
+
+[[section-valueextraction-builtinvalueextractors]]
+=== Built-in value extractors
+
+Hibernate Validator comes with built-in value extractors for the usual Java container
+types so, apart if you are using your own custom container types (or the ones
+of external libraries such as Guava's `Multimap`), you should not have to add your
+own value extractors.
+
+Built-in value extractors are present for all the following container types:
+
+ * `java.util.Iterable`;
+ * `java.util.List`;
+ * `java.util.Map`: for keys and values;
+ * `java.util.Optional`, `java.util.OptionalInt`, `java.util.OptionalLong` and `java.util.OptionalDouble`;
+ * {javafxDocsUrl}[JavaFX]'s `ObservableValue` (see <<section-valueextraction-javafx>>
+   for more details).
+
+The complete list of built-in value extractors with all the details on how they
+behave can be found in the
+{bvSpecUrl}#valueextractordefinition-builtinvalueextractors[Bean Validation specification].
+
+=== Implementing a `ValueExtractor`
+
+To extract values from a custom container, one needs to implement a `ValueExtractor`.
```

Comment:
Shouldn't you mention from the start that implementing it is not enough, and it needs to be registered afterwards?